docs/index.rst

```diff
@@ -431,6 +493,125 @@ A dialogue can be archived by setting ``is_archived`` to ``true`` when :ref:`rep
 
 .. TODO ^ reference events if and when they are documented
 
+.. _revisions:
+
+Revisions
+---------
+
+.. http:get:: /projects/(str:project_id)/dialogues/(str:dialogue_id)/revisions/
+
+  Retrieves the :ref:`revisions <data-revisions>` of dialogue ``dialogue_id``
+  in the project ``project_id``.
+
+  .. sourcecode:: http
+
+     GET /projects/23/dialogues/21/revisions/ HTTP/1.1
+
+  .. sourcecode:: http
+
+     HTTP/1.1 200 OK
+     Content-Type: application/json
+
+     [{
+       "id": "1",
+       "user_id": "17",
+       "created": 1459943775033,
+       "type": "edit",
+       "details": {
+         "id": "start",
+         "title": "Start of sequence",
+       },
+       "properties": {
+         "edit_type": "new_sequence",
+         "patch": {
+           "op": "add",
+           "path": "/sequences",
+           "value": {
+             "id": "start",
+             "title": "Start of sequence",
+             "blocks": []
+           }
+         }
+       }
+     }]
+
+  :query number page:
+    1-based index of the page of revisions to show. Defaults to ``1``.
+  :query number per_page:
+    Number of revisions to show per page. Defaults to ``30``. Maximum is
+    ``100``.
+  :query string ordering:
+    The ordering of the returned revisions. ``created`` returns the revisions
+    in ascending order of creation date and ``-created`` returns the revisions
+    in descending order of creation date. Defaults to ``-created``.
+
+
+.. http:post:: /projects/(str:project_id)/dialogues/(str:dialogue_id)/revisions/
```

Comment:
Yep. The UI will send over sets of revisions rather than sending over a dialogue description. For example, we might record one revision for moving a block and another revision for editing a block. We want to send those changes over as separate revisions so that revisions correspond to a meaningful set of changes rather than representing the state of a dialogue at a particular instance of time.